Bring back the `from` parameter on Connect buttons

ryelle opened the pull request:
This PR adds a new prop to `ConnectButton`, which lets developers pass in `from`, which will identify where in the app a button is (a descriptive identifier). The prop is not required, and should be a string.

The prop is added to the connection URL, where we'll track it as part of the connection flow to WordPress.com.

To test:

- With PR applied to a disconnected Jetpack site, visit the Jetpack dashboard
- Click/inspect the "Connect Jetpack" button's `href`
- The top button's href should end with `&from=landing-page-top`
- The bottom button's href should end with `&from=landing-page-bottom`

cc @jessefriedman @dereksmart

dereksmart commented:
These changes look good, however do we also want to add these to the user link buttons?  (@jessefriedman)

There is a connect URL built in the [unlinked user notice](https://github.com/Automattic/jetpack/blob/master/_inc/client/components/jetpack-notices/index.jsx#L177), as well as the [connection settings for unlinked users](https://github.com/Automattic/jetpack/blob/master/_inc/client/components/module-settings/index.jsx#L578).

dereksmart commented on _inc/client/components/connect-button/index.jsx:
```diff
@@ -25,12 +25,14 @@ const ConnectButton = React.createClass( {
 	displayName: 'ConnectButton',
 
 	propTypes: {
-		type: React.PropTypes.bool
```
Not sure why this was here :)

ryelle commented:
I added the `from` to the other connection URLs I found - the names are arbitrary (before I was matching the previous location labels), so they can be changed if we want.

I also added another parameter passed into the connection flow, `additional-user`, which we can also track for second+ user connections.

jessefriedman commented:
Very happy to see this coming back, thank you all.

dereksmart commented:
LGTM! thanks @ryelle
